src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -74,7 +74,7 @@ public GlobbingUrlBuilder(IFileProvider fileProvider, IMemoryCache cache, PathSt
         /// <param name="includePattern">The file globbing include pattern.</param>
         /// <param name="excludePattern">The file globbing exclude pattern.</param>
         /// <returns>The list of URLs</returns>
-        public virtual IEnumerable<string> BuildUrlList(string staticUrl, string includePattern, string excludePattern)
+        public virtual ICollection<string> BuildUrlList(string staticUrl, string includePattern, string excludePattern)
```

Comment:
ewww `ICollection<T>` :stuck_out_tongue_closed_eyes: - I think it's fine for a public API, just don't be surprised if you change it again later.

Do you then do a `foreach` over the collection?